stmt/stmts/grant.go

```diff
@@ -0,0 +1,760 @@
+		}
+		_, err = st.Exec(ctx)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+// Compose update stmt assignment list for global scope privilege update.
+func composeGlobalPrivUpdate(priv mysql.PrivilegeType) ([]expression.Assignment, error) {
+	if priv == mysql.AllPriv {
+		assigns := []expression.Assignment{}
+		for _, v := range mysql.Priv2UserCol {
+			a := expression.Assignment{
+				ColName: v,
+				Expr:    expression.Value{Val: "Y"},
+			}
+			assigns = append(assigns, a)
+		}
+		return assigns, nil
+	}
+	col, ok := mysql.Priv2UserCol[priv]
+	if !ok {
+		return nil, errors.Errorf("Unknown priv: %s", priv)
+	}
+	asgn := expression.Assignment{
+		ColName: col,
+		Expr:    expression.Value{Val: "Y"},
+	}
+	return []expression.Assignment{asgn}, nil
+}
+
+// Compose update stmt assignment list for db scope privilege update.
+func composeDBPrivUpdate(priv mysql.PrivilegeType) ([]expression.Assignment, error) {
+	if priv == mysql.AllPriv {
+		assigns := []expression.Assignment{}
+		for _, p := range mysql.AllDBPrivs {
+			v, ok := mysql.Priv2UserCol[p]
+			if !ok {
+				return nil, errors.Errorf("Unknown db privilege %s", priv)
+			}
+			a := expression.Assignment{
+				ColName: v,
+				Expr:    expression.Value{Val: "Y"},
+			}
+			assigns = append(assigns, a)
+		}
+		return assigns, nil
+	}
+	col, ok := mysql.Priv2UserCol[priv]
+	if !ok {
+		return nil, errors.Errorf("Unknown priv: %s", priv)
+	}
+	asgn := expression.Assignment{
+		ColName: col,
+		Expr:    expression.Value{Val: "Y"},
+	}
+	return []expression.Assignment{asgn}, nil
+}
+
+// Compose update stmt assignment list for table scope privilege update.
+func composeTablePrivUpdate(ctx context.Context, priv mysql.PrivilegeType, name string, host string, db string, tbl string) ([]expression.Assignment, error) {
+	var (
+		newTablePriv  string
+		newColumnPriv string
+	)
+	if priv == mysql.AllPriv {
+		for _, p := range mysql.AllTablePrivs {
+			v, ok := mysql.Priv2SetStr[p]
+			if !ok {
+				return nil, errors.Errorf("Unknown table privilege %s", p)
+			}
+			if len(newTablePriv) == 0 {
+				newTablePriv = v
+			} else {
+				newTablePriv = fmt.Sprintf("%s,%s", newTablePriv, v)
+			}
+		}
+		for _, p := range mysql.AllColumnPrivs {
+			v, ok := mysql.Priv2SetStr[p]
+			if !ok {
+				return nil, errors.Errorf("Unknown column privilege %s", p)
+			}
+			if len(newColumnPriv) == 0 {
+				newColumnPriv = v
+			} else {
+				newColumnPriv = fmt.Sprintf("%s,%s", newColumnPriv, v)
+			}
+
+		}
+	} else {
+		currTablePriv, currColumnPriv, err := getTablePriv(ctx, name, host, db, tbl)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		p, ok := mysql.Priv2SetStr[priv]
+		if !ok {
+			return nil, errors.Errorf("Unknown priv: %s", priv)
+		}
+		if len(currTablePriv) == 0 {
+			newTablePriv = p
+		} else {
+			newTablePriv = fmt.Sprintf("%s,%s", currTablePriv, p)
+		}
+		for _, cp := range mysql.AllColumnPrivs {
+			if priv == cp {
```

Comment:
p is string but cp and priv is PrivilegeType.
p is used to modify TypeSet column.